NorthstarDedicatedTest/dllmain.cpp

```diff
@@ -68,6 +78,114 @@ void WaitForDebugger(HMODULE baseAddress)
 	}
 }
 
+void freeLibrary(HMODULE hLib)
+{
+	bool freed = FreeLibrary(hLib);
+	if (!freed)
+	{
+		spdlog::error("There was an error while trying to free library");
+	}
+}
+
+bool LoadPlugins()
+{
+
+	std::vector<fs::path> paths;
+
+	std::string pluginPath = GetNorthstarPrefix() + "/plugins";
+
+	// ensure dirs exist
+	fs::recursive_directory_iterator iterator(pluginPath);
+	if (std::filesystem::begin(iterator) == std::filesystem::end(iterator))
+	{
+		spdlog::warn("Could not find any plugins. Skipped loading plugins");
+		return false;
+	}
+	for (auto const& entry : iterator)
+	{
+		if (fs::is_regular_file(entry) && entry.path().extension() == ".dll")
+			paths.emplace_back(entry.path().filename());
+	}
+	// system("pause");
+	initGameState();
+	// spdlog::info("Loading the following DLLs in plugins folder:");
+	for (fs::path path : paths)
+	{
+		std::string pathstring = (pluginPath / path).string();
+		std::wstring wpath = (pluginPath / path).wstring();
+
+		LPCWSTR wpptr = wpath.c_str();
+		HMODULE datafile =
+			LoadLibraryExW(wpptr, 0, LOAD_LIBRARY_AS_DATAFILE | LOAD_LIBRARY_AS_IMAGE_RESOURCE); // Load the DLL as a data file
+		if (datafile == NULL)
+		{
+			spdlog::info("Failed to load library {}: ", std::system_category().message(GetLastError()));
+			continue;
+		}
+		HRSRC manifestResource = FindResourceW(datafile, MAKEINTRESOURCE(101), MAKEINTRESOURCE(RT_RCDATA));
+
+		if (manifestResource == NULL)
+		{
+			spdlog::info("Could not find manifest for library {}", pathstring);
+			freeLibrary(datafile);
+			continue;
+		}
+		spdlog::info("Loading resource from library");
+		HGLOBAL myResourceData = LoadResource(datafile, manifestResource);
+		if (myResourceData == NULL)
+		{
+			spdlog::error("Failed to load resource from library");
+			freeLibrary(datafile);
+			continue;
+		}
+		int manifestSize = SizeofResource(datafile, manifestResource);
+		const char* manifestBinaryData = (const char*)LockResource(myResourceData);
+
+		std::string manifest = std::string(manifestBinaryData, 0, manifestSize);
```

Comment:
You can initialize it like `std::string manifest(manifestBinaryData, 0, manifestSize);`. I also recommend doing `freeLibrary(datafile);` after this then removing the occurrences afterwards (the string constructor copies).